pkg/controllers/cliconfig/cli_config_controller.go

```diff
@@ -0,0 +1,466 @@
+	routeclient "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/cluster-authentication-operator/pkg/controllers/configobservation"
+)
+
+var (
+	scheme  = runtime.NewScheme()
+	codecs  = serializer.NewCodecFactory(scheme)
+	encoder = codecs.LegacyCodec(osinv1.GroupVersion) // TODO I think there is a better way to do this
+)
+
+func init() {
+	utilruntime.Must(osinv1.Install(scheme))
+}
+
+// knownConditionNames lists all condition types used by this controller.
+// These conditions are operated and defaulted by this controller.
+// Any new condition used by this controller sync() loop should be listed here.
+var knownConditionNames = sets.NewString(
+	"CLIConfigDegraded",
+	"OAuthConfigDegraded",
+	"AuthCLIConfigProgressing",
+)
+
+type cliConfigController struct {
+	services       corev1client.ServicesGetter
+	secrets        corev1client.SecretsGetter
+	configMaps     corev1client.ConfigMapsGetter
+	operatorClient v1helpers.OperatorClient
+	route          routeclient.RouteInterface
+	auth           operatorv1client.AuthenticationsGetter
+}
+
+func NewCLIConfigController(kubeInformersForTargetNamespace informers.SharedInformerFactory, secrets corev1client.SecretsGetter, configMaps corev1client.ConfigMapsGetter, services corev1client.ServicesGetter,
+	operatorClient v1helpers.OperatorClient, authentication operatorv1client.AuthenticationsGetter, route routeclient.RouteInterface, recorder events.Recorder) factory.Controller {
+	c := &cliConfigController{
+		services:       services,
+		secrets:        secrets,
+		configMaps:     configMaps,
+		operatorClient: operatorClient,
+		auth:           authentication,
+		route:          route,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Informer(),
+		operatorClient.Informer(),
+	).ResyncEvery(30*time.Second).WithSync(c.sync).ToController("CLIConfig", recorder.WithComponentSuffix("cli-config-controller"))
+}
+
+func (c *cliConfigController) getRoute(ctx context.Context) (*routev1.Route, []operatorv1.OperatorCondition) {
+	// route is a pre-requirement for this sync
+	// it is created by metadata controller
+	// if route does not exists, do nothing and wait
+	route, err := c.route.Get(ctx, "oauth-openshift", metav1.GetOptions{})
+	if err != nil {
+		return nil, []operatorv1.OperatorCondition{
+			{
+				Type:    "CLIConfigDegraded",
```

Comment:
This isn't related to the CLI.  You want this degraded message to point to where the customer should look for a fix or what the impact of the failure is.  In this case, IngressCreationDegraded?

Reply:
@deads2k the result of this controller sync loop is `"v4-0-config-system-cliconfig`

Reply:
i mean i'm open for suggestions of the name for this controller :-)